struct.go

```diff
@@ -424,6 +438,15 @@ func reflectWithProperType(t reflect.Type, key *Key, field reflect.Value, delim
 		key.SetValue(fmt.Sprint(field.Interface().(time.Time).Format(time.RFC3339)))
 	case reflect.Slice:
 		return reflectSliceWithProperType(key, field, delim, allowShadow)
+	case reflect.Ptr:
+		isBoolPtr := t.Elem().Kind() == reflect.Bool
+		if isBoolPtr {
+			if !field.IsNil() {
+				key.SetValue(fmt.Sprint(field.Elem().Bool()))
+			}
+		} else {
+			return fmt.Errorf("unsupported type '%s'", t)
+		}
```

Comment:
See [the other comment](https://github.com/go-ini/ini/pull/200/files#r309976890).